### DISCUSSION TITLE
securecrt and teleport.

### DISCUSSION THREAD

**Discussion by johandeclercqdemocon** [gravitational/teleport#24965] [Q&A] [answered]
Hi, 
how can I use teleport with securecrt on ubuntu ? 
The thing is I need to connect to a cluster and then from there do ssh to a host. 
Is there somebody who managed to do that already ? 
I tried with tunneling but I don't get authenticated.   Issue is that connectoin to the proxy and then to the cluster is from terminal => not global to the system.   So how can I then open an ssh connection to the intended host via secure crt ?

**Comment by webvictim** [comment]
I don't believe SecureCRT supports SSH host certificates which would be required for it to work with Teleport.

We do have support for PuTTY (which does support all the necessary certificate types) coming soon.

**Comment by johandeclercqdemocon** [comment]
Can you elaborate a bit more on SSH host certificates ?
Then I will open a feature request at Van Dyke.

Op ma 24 apr 2023 om 17:18 schreef Gus Luxton ***@***.***>:

**Comment by johandeclercqdemocon** [comment]
Is it this that you mean ?
SSH certificate support - VanDyke Software Forums
<https://forums.vandyke.com/showthread.php?t=13980>

Op ma 24 apr 2023 om 17:18 schreef Gus Luxton ***@***.***>:

**Reply by webvictim** [reply]
Yes, that's the one.

**Comment by johandeclercqdemocon** [comment]
Is this the functionality on putty that you are refering to.

https://www.chiark.greenend.org.uk/~sgtatham/quasiblog/putty-certificates/

When this is implemented in putty, then I can put speed up things at van 
dyke.

**Reply by webvictim** [reply]
Yes. This is already implemented in PuTTY 0.78 and works today with Teleport.

**Comment by Atroskelis** [comment]
Hello, sorry for bumping this but did you ever get it to work? I can complete the proxy(firewall action) but not the certificate part.

**Reply by webvictim** [reply]
Pretty sure there's been no movement on this. You'd have to check in directly with the authors (Van Dyke software)

**Reply by Atroskelis** [reply]
> Hello, sorry for bumping this but did you ever get it to work? I can complete the proxy(firewall action) but not the certificate part.

@johandeclercqdemocon I've contacted Van Dyke Software about this and they gave me a pre-release of SecureCRT version 9.6.0 where Teleport works, previous versions do not work as SSH CA certs are not accepted. 

Support said that it usually takes a maximum 3-4 months for them to release the official 9.6.0. You can enable mail notifications to get notified.

**Comment by johandeclercqdemocon** [comment]
My customer dropped teleport. Anyway let me know how it goes.

**Comment by whatthejay** [comment]
@Atroskelis Could you please share more detail on how you made it work? I got the pre-release 9.6.0.0 from Vandyke support for evaluation, but it is not working for me.

Here's how I did the steps:

1. Configure firewall with "Proxy command" type in SCRT to execute tsh.exe login --proxy={my_proxy} --auth={my_auth} --login={my_login_name} {hostname}

2. Configure Credentials to use the identity file automatically generated by Teleport

However connection fails with message below:

`[LOCAL] : Stream has closed [CLOSE_TYPE_NONSPECIFIC] : The client has disconnected from the server.  Reason: Unable to authenticate using any of the configured authentication methods.  `

**Reply by whatthejay** [reply]
By the way, there are a few pub/priv keys under the directory: 
C:\Users\{username}\.tsh\keys\{teleport-proxy-hostname}\

Which key is valid for SecureCRT? I tried all of them but nothing worked.

**Reply by Atroskelis** [reply]
Hello! I did try a lot of methods. Securecrt does not digest easily most stuff in the .tsh folder.

You can try using a tool to extract the private key from the ppk in a format securecrt likes 
Also try viewing trace logs when you connect. Basically you need the ppk for private and the generic pub in the proxy folder

Ultimately the machine which I tested with is in service, and ill only have aceess to it next week

In the end and even now ( with royal ts) im unable to make connections work without tsh puttyconfig on all jumps. Never knew why

**Reply by webvictim** [reply]
@whatthejay Your proxy command should be:

```
tsh.exe --proxy={my_proxy} --auth={my_auth} --login={my_login_name} proxy ssh {hostname}
```

You will likely need to configure other settings in order to make SecureCRT trust Teleport-issued host certificates, however.

I don't have access to SecureCRT at all (let alone the pre-release build with support for certificates) so I can't currently be much help.

@Atroskelis `tsh puttyconfig` does a number of things when run, but the most important here is probably that it adds the public key of Teleport's host CA (used to sign the certificates issued to hosts running Teleport) to PuTTY's registry key, and also updates the list of hostnames that the public key should be trusted for connecting to. This is needed because PuTTY uses an "allowlist" of authorised hostnames and won't permit connections to any host not added to this list.

**Reply by brantxs** [reply]
I'm using the Mac version 9.6.2, and everything works fine, but the Windows version 9.6.2 keeps reporting certificate verification errors.

**Reply by Atroskelis** [reply]
Hello everyone.

I managed to get a copy of SecureCRT for 9.6.3 and here are the steps i made.

- You have to be logged in to teleport before hand via some tsh login script or similar (i.e tsh ls must work)
- In the C:/Users/yourname/.tsh/keys/yourcluster are some files
- You must copy the C:/Users/yourname/.tsh/keys/yourcluster/yourname-ssh/cluster pub file to C:/Users/yourname/.tsh/keys/yourcluster
- You must rename your private key surname.name to yourcluster-cert with no extension.
This is because Securecrt expects a private key with the same name, i.e a pub file must always have a private key without .pub extension
- In securecrt you must create local firewall that runs your tsh proxy command, this gets attached to your session
- Then you can create your session, hosts and user are self explanatory, in the Authentication -> Publickey -> you must attach the pub file copied previously that has the private key with it.

Connect and test. No puttyconfig is required apparently.

**Comment by rumplestiltsman** [comment, marked as answer]
Support for OpenSSH Trusted Certificates as SSH host keys was added to SecureCRT/SecureFX version 9.6, which was released in November, 2024.

**OpenSSH Trusted Certificates as Host Keys**
Individual host certificates can be accepted/saved, and a signing cert for an entire domain can be added in the **SSH Host Keys** category of SecureCRT's Global Options.

![SCRT_9 6+_ImportOpenSSHCert(Teleport)_HostKey](https://github.com/user-attachments/assets/5596adbf-b673-40ea-8942-b5e175b661f2)


**OpenSSH Trusted Certificates for User Authentication (Public-Key)**
For user authentication with an OpenSSH Trusted Certificate:
- Configure the **Public Key Properties** in your SecureCRT saved session to point to the _private_ key file (OpenSSH private key file format). If all you have is a `.ppk` private key file, you'll need to convert a copy of the `.ppk` you have to OpenSSH format.
- Copy the `-cert.pub` file to the same directory where the private key resides.
- Rename the `-cert.pub` file to have the same base file name as the private key file. For example, if the private key file is named `mykey`, then the corresponding trusted certificate should be named `mykey-cert.pub` (and `mykey-cert.pub` should exist in the same directory where `mykey` exists)

![SCRT_User_Authentication-OpenSSH-Trusted-Certificate_PublicKey](https://github.com/user-attachments/assets/854337aa-51dd-4608-9d36-35a48f05d2b0)

**Reply by Atroskelis** [reply]
Very nice documentation! Congrats! 👏👏

**Reply by Atroskelis** [reply]
To add, apparently in the recent 9.7 beta you wont need to create an identity anymore  https://www.vandyke.com/products/beta/securecrt/new-in.html

**Comment by Atroskelis** [comment]
Hello! 

2026 Update, if you use Teleport VNET which gets rid of the burden of using tsh firewall.

Besides adding the CA as in the image above, the cert to use in the SSH2 Session config is in

C:\Users\<username>\AppData\Roaming\Teleport Connect\tsh\id_vnet

However, it seems vnet creates a new host session key upon connecting that means you will forever have to accept the destination host key.

EDIT 2026:

You can set it
up using the following edit to the SSH2.ini file that lives
in the Configuration folder SecureCRT is currently
configured to use:

D:"Automatically Accept Host Keys"=00000001

This does exactly what it says. It "Automatically accepts
host keys".

**Comment by Atroskelis** [comment]
Be advised, Lastest-ish SecureCRT is not working with teleport at the moment:

Response from SecureCRT support:

(this was with tsh proxy method)
The two related issues are:

  1) The SSH2 server to which you are attempting to
     authenticate does not accurately implement RFC 8308,
     which indicates that an SSH2 server sends a list of all
     the supported key signature algorithms that it
     supports/allows/implements in its server-sig-algs info.

     You can see from the above output that the SSH2 server
     does not indicate support for any
     *-[cert-v01@openssh.com](mailto:cert-v01@openssh.com) signature algorithms.

  2) SecureCRT 9.7.x introduced stricter adherence to the
     algorithms listed in an SSH2 server's RFC 8308
     server-sig-algs information. The problem was that
     SecureCRT was attempting to authenticate with key types
     that an SSH2 server did not support (and did not list
     in its server-sig-algs), which was resulting in locked
     devices/accounts for a number of customer scenarios. It
     was discovered that not even OpenSSH SSH2 server
     implementations accurately implement RFC 8308 (as they
     don't include any *-[cert-v01@openssh.com](mailto:cert-v01@openssh.com) signature
     algorithms in their server-sig-algs output, so an
     exception was implemented in SecureCRT version 9.7.1 as
     seen in the history file here:
